### PR TITLE
Add best practices for CPU backend doc

### PR DESCRIPTION
**Pull request by ZailiWang**
Content same as #103948 
@svekars the PR content is updated per your comment, but when trying to solve the conflict the original PR was closed by a mis-operation. Would you help handle this new one? sorry for the inconvenience.

### CODE REVIEW

File: docs/source/compile/best-practices-for-backends.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Best Practices for Backends
+===========================
+
+x86 CPU
+-------
+
+Compiled workloads on modern x86 CPUs are usually optimized by Single Instruction Multiple Data (SIMD) instruction sets. SIMD is a typical parallel processing technique for high performance computing, such as deep learning model training and inference. With SIMD applied, each compute unit performs the same instruction with different allocated data at any given time slot. The most commonly deployed x86 instruction set architectures (ISAs) enabling SIMD include `AVX, AVX2, AVX-512 <https://en.wikipedia.org/wiki/Advanced_Vector_Extensions>`_ and `AMX <https://en.wikipedia.org/wiki/Advanced_Matrix_Extensions>`_.
+
+You can check supported ISAs for your machine by using the `collect_env script <https://github.com/pytorch/pytorch/blob/main/torch/utils/collect_env.py>`_. As the script provides complete environment information for PyTorch, we can use ``grep`` to extract the line containing ISA information:
```
Comment (ZailiWang) [inline docs/source/compile/best-practices-for-backends.rst:9]: did not spell out ISA here as there was one above (in last sentence in paragraph 1)

**Review by svekars — APPROVED** [review]
LGTM!

**Pull request closed without merging**